### ISSUE TITLE
Port torch/csrc/jit/runtime/register_prim_ops_c10.cpp to new operator registration API

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#37348]
We have a new operator registration API introduced in #36258, and we need to port all use sites of the old registration API to use it.

Here is the general recipe for porting:
1. We must add def()s of the operators to the centralized location.  For `aten::` namespace, this location is the `TORCH_LIBRARY(aten, m) {` block in `aten/src/ATen/templates/TypeDefault.cpp`.  A def looks like:

```
m.def("aten::splitlines(str self, bool keepends=False) -> str[]");
```

2. We must translate the registrations in register_string_ops.cpp to the new registration form (staying in this file). You do this by adding a new `TORCH_LIBRARY_IMPL` block as such:

```
TORCH_LIBRARY_IMPL(aten, CatchAll, m) {
  m.impl("splitlines", [](std::string string, bool keepends) {
    // previous implementation
  });
  // ... rest of the registrations ...
}
```

A previous example: #37008

This task is reserved for bootcamp, please get in contact with me before starting work on it.

cc @suo

**Comment by RockingJavaBean** [comment]
@ezyang I would like to work on this bootcamp issue.

**Comment by ezyang** [comment]
@RockingJavaBean  I think Mohammad Mahdi Javanmard was planning to work on this. However, there are some other files which also need porting. Do you want to handle the registrations in torch/csrc/jit/runtime/register_distributed_ops.cpp perhaps?

**Comment by RockingJavaBean** [comment]
@ezyang Sure, I like working on the registrations in `register_distributed_ops.cpp`, and a new issue #37579 has been created for the port.
